soundata/datasets/fsd50k.py

```diff
@@ -0,0 +1,478 @@
+
+    ground_truth_dict = {}
+    with open(data_path, "r") as fhandle:
+        reader = csv.reader(fhandle, delimiter=",")
+        next(reader)
+        for line in reader:
+            if len(line) == 3:
+                ground_truth_dict[line[0]] = {
+                    "tags": list(line[1].split(",")) if "," in line[1] else line[1],
+                    "mids": list(line[2].split(",")) if "," in line[2] else line[2],
+                    "split": None,
+                }
+            if len(line) == 4:
+                if "," in line[2]:
+                    ground_truth_dict[line[0]] = {
+                        "tags": list(line[1].split(",")) if "," in line[1] else line[1],
+                        "mids": list(line[2].split(",")) if "," in line[2] else line[2],
+                        "split": line[3],
+                    }
+
+    return ground_truth_dict
+
+
+def load_fsd50k_vocabulary(data_path):
+    """
+    TODO
+    """
+    if not os.path.exists(data_path):
+        raise FileNotFoundError(
+            "FSD50K vocabulary file not found. Did you run .download()?"
+        )
+
+    fsd50k_to_audioset = {}
+    audioset_to_fsd50k = {}
+    with open(data_path, "r") as fhandle:
+        reader = csv.reader(fhandle, delimiter=",")
+        for line in reader:
+            fsd50k_to_audioset[line[1]] = line[2]
+            audioset_to_fsd50k[line[2]] = line[1]
+
+    return fsd50k_to_audioset, audioset_to_fsd50k
+
+
+@core.docstring_inherit(core.Dataset)
+class Dataset(core.Dataset):
+    """
+    The FSD50K dataset
+    """
+
+    def __init__(self, data_home=None):
+        super().__init__(
+            data_home,
+            name="fsd50k",
+            clip_class=Clip,
+            bibtex=BIBTEX,
+            remotes=REMOTES,
+            license_info=LICENSE_INFO,
+        )
+
+    @core.copy_docs(load_audio)
+    def load_audio(self, *args, **kwargs):
+        return load_audio(*args, **kwargs)
+
+    @core.copy_docs(load_ground_truth)
+    def load_ground_truth(self, *args, **kwargs):
+        return load_ground_truth(*args, **kwargs)
+
+    @core.copy_docs(load_fsd50k_vocabulary)
+    def load_fsd50k_vocabulary(self, *args, **kwargs):
+        return load_fsd50k_vocabulary(*args, **kwargs)
+
+    @property
+    def fsd50k_to_audioset(self):
+        data_path = os.path.join(
+            self.data_home, "FSD50K.ground_truth", "vocabulary.csv"
+        )
+        return load_fsd50k_vocabulary(data_path)[0]
+
+    @property
+    def audioset_to_fsd50k(self):
+        data_path = os.path.join(
+            self.data_home, "FSD50K.ground_truth", "vocabulary.csv"
+        )
+        return load_fsd50k_vocabulary(data_path)[1]
+
+    @property
+    def class_info(self):
+        data_path = os.path.join(
+            self.data_home, "FSD50K.metadata", "class_info_FSD50K.json"
+        )
+        return json.load(open(data_path, "r")) if os.path.exists(data_path) else None
```

Comment:
what is this one doing?

Reply:
Is basically a high-level dictionary of information about the labels. Basically, it provides a brief description of the label, and it lists some clip examples and verification examples of the label. That is why I am also declaring it here, at the dataset level.